backend/src/middleware/userMiddleware.js

```diff
@@ -11,6 +13,10 @@ export default {
       return result
     },
     UpdateUser: async (resolve, root, args, context, info) => {
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-17T14:12:51Z" title="Wednesday, April 17th 2019, 4:12:51 pm +02:00">Apr 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -11,6 +13,10 @@ export default {
       return result
     },
     UpdateUser: async (resolve, root, args, context, info) => {
```


This should go into a middleware, not a resolver. Ask @mattwr18 where he placed the validation for `SocialMedia`. We could have a nice `middleware/validations/index.js` where we place all data validations.

If you do it that way, you can also re-use the same validation method for `CreateUser` and `UpdateUser`.

Reply:
> [<img alt="ulfgebhardt" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-04-19T10:15:43Z" title="Friday, April 19th 2019, 12:15:43 pm +02:00">Apr 19, 2019</time>_

---


Do you refere to this? Or are there some other changes somewhere
![image](https://user-images.githubusercontent.com/1238238/56420003-98d49680-629c-11e9-9da5-a6a8a5be3a68.png)
`middleware/validations/index.js` is not existent in backend master
@roschaefer @mattwr18

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-19T11:55:42Z" title="Friday, April 19th 2019, 1:55:42 pm +02:00">Apr 19, 2019</time>_

---


Oh yes! Apparently this got overlooked. Either we use [constraint directives](https://github.com/confuser/graphql-constraint-directive) or we implement all data validations inside a middleware.

@ulfgebhardt could you give the constraint directives a try? You are the GraphQL expert and @mattwr18 had a hard time when he gave it a shot.